virttest/qemu_devices/qcontainer.py

```diff
@@ -2057,8 +2061,12 @@ def define_hbas(qtype, atype, bus, unit, port, qbus, pci_bus, iothread,
                                  "on %s by -blockdev." % (opt, name))
             if media == 'cdrom':
                 readonly = 'on'
-            protocol_node.set_param('read-only', readonly, bool)
             format_node.set_param('read-only', readonly, bool)
+
+            protocol_node.set_param('auto-read-only',
+                                    image_auto_readonly, bool)
+            protocol_node.set_param('discard', image_discard)
```

Comment:
From original codes: https://github.com/avocado-framework/avocado-vt/blob/d3a00367ff404d65f4a3e9ae106cf55c9c80c93a/virttest/qemu_devices/qcontainer.py#L2208
```
        if drv_extra_params:
            drv_extra_params = (_.split('=', 1) for _ in
                                drv_extra_params.split(',') if _)
            for key, value in drv_extra_params:
                if Flags.BLOCKDEV in self.caps:
                    if key == 'discard':
                        value = re.sub('on', 'unmap', re.sub('off', 'ignore', value))
```
What's the priority between `image_discard` and the above parameters? or which one will work if both of them are set?

Reply:
discard set in drv_extra_params will take effect because it is set specified for some test cases.
The change in this patch just handle the "default" situation, i.e. when we never set the params, qemu-kvm should include these settings in blockdev command line; Once, we set them to some values other than the ones in base-internal.cfg(please refer to https://gitlab.cee.redhat.com/kvm-qe/kar/-/merge_requests/504), the new settings should take effect.